### DISCUSSION TITLE
cronAdd not defined within a hook

### DISCUSSION THREAD

**Discussion by bberenberg** [pocketbase/pocketbase#5061] [Q&A] [answered]
My code is:

```
onAfterBootstrap((e) => {
    console.log('App initialized!');

    // Schedule the weather check to run daily at midnight
    cronAdd("weatherCheck", "0 0 * * *", () => { // Adjust cron schedule as needed
        checkWeather(e.app);
    });

    // Run the first check immediately
    checkWeather(e.app);
});
```

And on start I get `ReferenceError: cronAdd is not defined at <eval>:5:12(10)`

This is on PocketBase v0.22.13 running on an M1 Mac.

**Comment by ganigeorgiev** [comment, marked as answer]
`cronAdd` and `cronRemove` are only available at the top-level (this is an implementation detail how the JSVM works - https://github.com/pocketbase/pocketbase/discussions/2825#discussioncomment-6376210).

**Reply by bberenberg** [reply]
Thank you very much for the quick response.
